kanidmd/src/lib/interval.rs

```diff
@@ -24,4 +33,88 @@ impl IntervalActor {
             }
         });
     }
+
+    pub fn start_online_backup(
+        server: &'static QueryServerReadV1,
+        cfg: &OnlineBackup,
+    ) -> Result<(), ()> {
+        let outpath = cfg.path.to_owned();
+        let schedule = cfg.schedule.to_owned();
+        let versions = cfg.versions;
+
+        // Cron expression handling
+        let cron_expr = schedule.as_str().parse::<CronExpr>().map_err(|e| {
+            error!("Online backup schedule parse error: {}", e);
+        })?;
+
+        info!(
+            "Online backup schedule parsed as: {}",
+            cron_expr.describe(English::default())
+        );
+
+        if !Cron::new(cron_expr.clone()).any() {
+            error!(
+                "Online backup schedule error: '{}' will not match any date.",
+                schedule
+            );
+            return Err(());
+        }
+
+        // Output path handling
+        let op = Path::new(&outpath);
+
+        // does the path exist and is a directory?
+        if !op.exists() {
+            info!(
+                "Online backup output folder '{}' does not exist, trying to create it.",
+                outpath
+            );
+            fs::create_dir_all(&outpath).map_err(|e| {
+                error!(
+                    "Online backup failed to create output directory '{}': {}",
+                    outpath.clone(),
+                    e
+                )
+            })?;
+        }
+
+        if !op.is_dir() {
+            error!("Online backup output '{}' is not a directory or we are missing permissions to access it.", outpath);
+            return Err(());
+        }
+
+        // checking permissions (not sure about this)
```

Comment:
I've just checked, the umask is set in the process startup so you don't need to worry about it.